data/UnitOfWork.js

```diff
@@ -0,0 +1,27 @@
+const TaskMemoryDAO = require('./TaskMemoryDAO');
+const TaskRepository = require('./TaskRepository');
+const CourseMemoryDAO = require('./CourseMemoryDAO');
+const CourseRepository = require('./CourseRepository');
+
+const validateTaskTitle = (title) => (title && (typeof title === 'string' || title instanceof String));
```

Comment:
I don't think you're ever going to get a String as an object unless you create one intentionally. the first type check should be enough.